diff --git a/node_modules/@nx/react/src/utils/versions.d.ts b/node_modules/@nx/react/src/utils/versions.d.ts
index 53d36d4..bb332ac 100644
--- a/node_modules/@nx/react/src/utils/versions.d.ts
+++ b/node_modules/@nx/react/src/utils/versions.d.ts
@@ -24,7 +24,7 @@ export declare const emotionReactVersion = "11.11.1";
 export declare const emotionBabelPlugin = "11.11.0";
 export declare const styledJsxVersion = "5.1.2";
 export declare const reactRouterDomVersion = "6.29.0";
-export declare const reactRouterVersion = "^7.2.0";
+export declare const reactRouterVersion = "^7.5.2";
 export declare const reactRouterIsBotVersion = "^4.4.0";
 export declare const testingLibraryReactVersion = "16.1.0";
 export declare const testingLibraryDomVersion = "10.4.0";
diff --git a/node_modules/@nx/react/src/utils/versions.js b/node_modules/@nx/react/src/utils/versions.js
index 1449021..8dd93e6 100644
--- a/node_modules/@nx/react/src/utils/versions.js
+++ b/node_modules/@nx/react/src/utils/versions.js
@@ -28,9 +28,10 @@ exports.emotionStyledVersion = '11.11.0';
 exports.emotionReactVersion = '11.11.1';
 exports.emotionBabelPlugin = '11.11.0';
 // WARNING: This needs to be in sync with Next.js' dependency or else there might be issues.
+// reactRouterVersion patched to 7.5.2 to address security issue
 exports.styledJsxVersion = '5.1.2';
 exports.reactRouterDomVersion = '6.29.0';
-exports.reactRouterVersion = '^7.2.0';
+exports.reactRouterVersion = '^7.5.2';
 exports.reactRouterIsBotVersion = '^4.4.0';
 exports.testingLibraryReactVersion = '16.1.0';
 exports.testingLibraryDomVersion = '10.4.0';
